pkg/repositories/transformers/node_execution.go

```diff
@@ -173,12 +173,19 @@ func UpdateNodeExecutionModel(
 	// Update TaskNodeMetadata, which includes caching information today.
 	if request.Event.GetTaskNodeMetadata() != nil {
 		st := request.Event.GetTaskNodeMetadata().GetCacheStatus().String()
-		nodeExecutionClosure.TargetMetadata = &admin.NodeExecutionClosure_TaskNodeMetadata{
+		targetMetadata := &admin.NodeExecutionClosure_TaskNodeMetadata{
 			TaskNodeMetadata: &admin.TaskNodeMetadata{
 				CacheStatus: request.Event.GetTaskNodeMetadata().GetCacheStatus(),
 				CatalogKey:  request.Event.GetTaskNodeMetadata().GetCatalogKey(),
 			},
 		}
+		if request.Event.GetTaskNodeMetadata().DynamicWorkflow != nil {
+			targetMetadata.TaskNodeMetadata.DynamicWorkflow = &admin.DynamicWorkflowNodeMetadata{
```

Comment:
are we storing it in the DB? Where do we store the current compiled workflow - i thought s3?

Reply:
oh yes good point! will revise to make this relational

Reply:
updated. also this follows the workflow model where a get will go and fetch the closure but the call to list node executions will not

Reply:
ok, but how does the lister know that the dynamic workflow has not been fetched?

Reply:
Also maybe I dont understand this, but where is the offloading code?

Reply:
@kumare3 the lister will b/c a dynamic workflow id will exist but the corresponding closure will not. offloading is in this refactored shared method: https://github.com/flyteorg/flyteadmin/pull/182/files#diff-8826e8af1b89cbbc5038dae18de4989e16125b6148e236cd535954dc6cfb69c9R94